c7n/resources/ecr.py

```diff
@@ -69,3 +80,76 @@ def _augment(r):
             resources = list(filter(None, w.map(_augment, resources)))
 
         return super(ECRCrossAccountAccessFilter, self).process(resources, event)
+
+
+@actions.register('remove-statements')
+class RemovePolicyStatement(BaseAction):
+    """Action to remove policy statements from ECR
+
+    :example:
+
+        .. code-block: yaml
+
+            policies:
+              - name: ecr-remove-cross-accounts
+                resource: s3
+                filters:
+                  - type: cross-account
+                actions:
+                  - type: remove-statements
+                    statement_ids: matched
+    """
+
+    schema = type_schema(
+        'remove-statements',
+        required=['statement_ids'],
+        statement_ids={'oneOf': [
+            {'enum': ['matched']},
+            {'type': 'array', 'items': {'type': 'string'}}]})
+    permissions = ("ecr:SetRepositoryPolicy",)
+
+    def process(self, resources):
+        with self.executor_factory(max_workers=3) as w:
+            futures = {}
+            results = []
+            for resource in resources:
+                futures[w.submit(self.process_registries, resource)] = resource
+            for f in as_completed(futures):
+                if f.exception():
+                    self.log.error('error modifying registry:%s\n%s',
+                                   resource['repositoryName'], f.exception())
```

Comment:
resource is not in scope, it will always print the last repo in the set of resources